code/game/machinery/doors/airlock.dm

```diff
@@ -320,203 +359,112 @@ About the new airlock wires panel:
 /obj/machinery/door/airlock/proc/shock_user(mob/user, prob)
 	return (!issilicon(user) && isElectrified() && shock(user, prob))
 
-/obj/machinery/door/airlock/update_icon(state=0, override=0)
+/obj/machinery/door/airlock/update_icon(updates=ALL, state=0, override=FALSE)
 	if(operating && !override)
 		return
-	check_unres()
-	icon_state = density ? "closed" : "open"
-	switch(state)
-		if(0)
-			if(density)
-				state = AIRLOCK_CLOSED
-			else
-				state = AIRLOCK_OPEN
+
+	if(!state)
+		state = density ? AIRLOCK_CLOSED : AIRLOCK_OPEN
+	airlock_state = state
+
+	. = ..()
+
+	if(hasPower() && unres_sides)
+		set_light(2, 1)
+	else
+		set_light(0)
+
+/obj/machinery/door/airlock/update_icon_state()
+	switch(airlock_state)
 		if(AIRLOCK_OPEN, AIRLOCK_CLOSED)
+			icon_state = ""
```

Comment:
huh? "" doesn't exist in `icons/obj/doors/airlocks/station/public.dmi` (nor in other airlock files), same as `nonexistenticonstate`.
Does that mean that base icon is *always* empty and everything is done via overlays??
If so we should.. probably remove the switch altogether, and set `icon_state=null` unconditionally?..

Reply:
I actually tried that, and it doesn't work somehow. I'm not sure what kind of weird wizard magic occurs here, but just setting the icon unconditionally to either `""`/`"nonexistenticonstate"` will break the animations. Likewise, replacing `""` with `null` will also break things. `Flick` just doesn't seem to play nicely otherwise.

I should leave a comment there for that.